pydantic/types.py

```diff
@@ -606,3 +607,68 @@ def _get_brand(card_number: str) -> PaymentCardBrand:
         else:
             brand = PaymentCardBrand.other
         return brand
+
+
+BYTE_SIZES = {
+    '': 1,
+    'B': 1,
+    'kB': 10 ** 3,
+    'MB': 10 ** 6,
+    'GB': 10 ** 9,
+    'TB': 10 ** 12,
+    'PB': 10 ** 15,
+    'EB': 10 ** 18,
+    'KiB': 2 ** 10,
+    'MiB': 2 ** 20,
+    'GiB': 2 ** 30,
+    'TiB': 2 ** 40,
+    'PiB': 2 ** 50,
+    'EiB': 2 ** 60,
+}
+BYTE_SIZES.update({k.lower(): v for k, v in BYTE_SIZES.items()})
+byte_string_re = re.compile(r'^\s*(\d*\.?\d+)\s*(\w+)?', re.IGNORECASE)
```

Comment:
```suggestion
byte_string_re = re.compile(r'^\s*(\d*\.?\d+)\s*(\w+)\s*', re.IGNORECASE)
```
`' 123   '` will be parsed to int anyway.

Reply:
`int(1.0)` will not be however.

Reply:
okay.